contracts/YCreditStrat.sol

```diff
@@ -0,0 +1,80 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity 0.7.3;
+
+import "./IStrat.sol";
+import "./IYCredit.sol";
+import "./IVault.sol";
+import "./Timelock.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+contract YCreditStrat is IStrat {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    IVault public vault;
+    IYCredit public yCredit;
+    Timelock public timelock;
+
+    modifier onlyVault {
+        require(msg.sender == address(vault));
+        _;
+    }
+
+    constructor(IVault vault_, Timelock timelock_) {
+        vault = vault_;
+        timelock = timelock_;
+        yCredit = IYCredit(address(vault_.underlying())); // vault underlying must be yCredit
+    }
+
+    function invest() external override onlyVault {
+        reinvest();
```

Comment:
We're staking and checking balances a redundant amount of times here. Is the cost worth the convenient internal function usage?

Reply:
You're right. I'll get rid of the function call and do it directly in each function